app/views/TaxYearSummary.scala.html

```diff
@@ -48,17 +48,24 @@
 @import implicitDateFormatter._
 
 @taxCalculationContent = @{
-    modelOpt.fold(ifEmpty = taxCalculationNotPresent)(calcOverview => taxCalculationOverview(calcOverview))
+    modelOpt.fold(ifEmpty = taxCalculationNotPresent)(calcOverview => taxCalculationOverview(calcOverview, isAgent))
 }
 
 @taxCalculationNotPresent = {
     @h2(msg = messages("tax-year-summary.tax-calculation.no-calc"), classes = "govuk-heading-l", optId=Some("no-calc-data-header"))
     @p(id=Some("no-calc-data-note"))(Html(messages("tax-year-summary.tax-calculation.no-calc.note")))
 }
 
-@taxCalculationOverview(model: TaxYearSummaryViewModel) = {
-    @if(model.messages.exists(_.errorMessages.size > 0) && !isAgent) {
-        @errorMessage(model)
+@getMessageKey(isAgent: Boolean) = {
+  @if(isAgent) {tax-year-summary.agent.message} else {tax-year-summary.message}
+}
+
+@taxCalculationOverview(model: TaxYearSummaryViewModel, isAgent: Boolean) = {
+    @if(model.messages.exists(_.errorMessages.size > 0)) {
+       @{
+            val key = getMessageKey(isAgent)
+            errorMessage(model,s"${key}".trim(), isAgent)
```

Comment:
you can move trim in getMessageKey function,

Reply:
I've removed the function and used a function literal instead. We dont need the trim() now.